packages/sanity/src/studio/screens/NotAuthenticatedScreen.tsx

```diff
@@ -1,15 +1,61 @@
-import {Flex, Card, Container} from '@sanity/ui'
-import React from 'react'
-import {WorkspaceAuth} from '../components/navbar/workspace'
+import {CurrentUser} from '@sanity/types'
+import {Button, Card, Dialog, Stack, Text} from '@sanity/ui'
+import React, {useCallback, useEffect, useState} from 'react'
+import {useActiveWorkspace} from '../activeWorkspaceMatcher'
 
 export function NotAuthenticatedScreen() {
+  const [currentUser, setCurrentUser] = useState<CurrentUser | null>(null)
+  const [error, handleError] = useState<unknown>(null)
+
+  if (error) throw error
+
+  const {activeWorkspace} = useActiveWorkspace()
+
+  const handleLogout = useCallback(() => {
+    activeWorkspace.auth.logout?.()
+  }, [activeWorkspace])
+
+  useEffect(() => {
+    activeWorkspace.auth.handleCallbackUrl?.().catch(handleError)
+  }, [activeWorkspace.auth])
+
+  useEffect(() => {
```

Comment:
Also  a missing unsubscribe in effect cleanup here.

Reply:
Thx 👍